File: lecture010/powerBtr.java
import java.util.Scanner;
public class powerBtr{
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args){
        System.out.println(powerBtr(2,5));
    }
    public static int powerBtr(int a , int b){
        if(b==0) return 1;
        
        int recAns=powerBtr(a,b/2);
        recAns*=recAns;
        if(b%2!=0){
            recAns*=a;
        }

        return recAns;
    }
}

File: lecture010/powerBtr.cpp
#include <iostream>

using namespace std;
int powerBtr(int a, int b)
{
    if (b == 0)
        return 1;
    //  cout<<"*b"<<endl;
    int recans = powerBtr(a, b / 2);
    recans *= recans;
    //cout<<"#b"<<endl;
    if (b % 2 != 0)
        recans *= a;

    return recans;
}
int main(int args, char **argv)
{
    cout << powerBtr(2, 5) << endl;
}